### PR TITLE
[WIP] Audio preprocessing tutorial.

### PR DESCRIPTION
**Pull request by vincentqb**
Initial tutorial with an overview of dataset loading and transformers. Created using the following [notebook](https://github.com/vincentqb/audio-tutorial/blob/master/Tutorial.ipynb). 

TODO

- [x]  Update as soon as the transformers have been standardized, see [PR](https://github.com/pytorch/audio/pull/152). For instance, we expect LC2CL to no longer be needed.
- [x] Confirm location to put tutorial
- [x] Confirm images are generated on the web

In a later iteration, we should demo a neural network implementation, maybe inspired by [a prior version](https://pytorch.org/tutorials/beginner/audio_classifier_tutorial.html?highlight=audio) (which no longer appears in the left hand side menu but is still searchable).

The initial pull request was done in [torchaudio](https://github.com/pytorch/audio/pull/150).

### CODE REVIEW

**Comment by vincentqb** [comment]
- [ ] Beginner or Intermediate?
- [ ] Add torchaudio package, change to master, and test in a new branch.
- [ ] Add audio signal to `_static/img/`
- [ ] Test using [test](https://github.com/brianjo/pttutorialtest)
- [ ] Delete audio old tutorial in separate PR

**Comment by vincentqb** [comment]
While testing with the mini-website, I'm running into failing nightlies from PyTorch.

@jamarshon -- do you know one that's compatible with torchaudio that works?

**Comment by brianjo** [comment]
Just FYI, seeing similar issues with Torchvision against master. We're working on this branch https://github.com/pytorch/tutorials/pull/575. Once I get tv fixed, I'll enable audio. Cheers.

**Comment by jamarshon** [comment]
try 1.2.0.dev20190722 or a few days back. torchaudio CI passed on those ones

**Comment by vincentqb** [comment]
Test build of branch with tutorial in #572. Thanks @brianjo!

**Comment by vincentqb** [comment]
I was able to successfully generate the page using the [mini-website](https://github.com/vincentqb/pttutorialtest) with the latest path using `1.2.0.dev20190725+cpu`, sox from conda, and the current version of torchaudio.

**Comment by vincentqb** [comment]
@brianjo -- do you know where the style sheet controlling the size of the images generated in the document? thanks!

File: beginner_source/audio_preprocessing_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+# -  **MelScale**: This turns a normal STFT into a Mel-frequency STFT,
+#    using a conversion matrix.
+# -  **SpectrogramToDB**: This turns a spectrogram from the
+#    power/amplitude scale to the decibel scale.
+# -  **MFCC**: Create the Mel-frequency cepstrum coefficients from a
+#    waveform.
+# -  **MelSpectrogram**: Create MEL Spectrograms from a waveform using the
+#    STFT function in PyTorch.
+# -  **MuLawEncoding**: Encode waveform based on mu-law companding.
+# -  **MuLawDeconding**: Decode mu-law encoded waveform.
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: Decoding

File: beginner_source/audio_preprocessing_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+# 
+
+# Compute median relative difference
+err = ((waveform-reconstructed).abs() / waveform.abs()).median()
+
+print("Median relative difference between original and MuLaw reconstucted signals: {:.2%}".format(err))
+
+
+######################################################################
+# Migrating to Torchaudio from Kaldi
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+# ----------------------------------
+# 
+# Users may be familiar with
+# `Kaldi <http://github.com/kaldi-asr/kaldi>`_, a toolkit for speech
+# recognition. Torchaudio offers compatibility with it in
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+# ``torchaudio.kaldi_io``. It can indeed read from kaldi scp, or ark file
+# or streams with:
+# 
+# -  read_vec_int_ark
+# -  read_vec_flt_scp
+# -  read_vec_flt_arkfile/stream
+# -  read_mat_scp
+# -  read_mat_ark
+# 
+# Torchaudio provides Kaldi-compatible transforms for ``spectrogram`` and
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+plt.figure()
+plt.imshow(fbank.transpose(0,1).numpy(), cmap='gray')
+
+
+######################################################################
+# Conclusion
+# ----------
+# 
+# We used an example raw audio signal, or waveform, to illustrate how to
+# open an audio file using Torchaudio, and how to pre-process and
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+# transform such waveform. Given that Torchaudio is built on PyTorch,
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 7]
```diff
@@ -0,0 +1,237 @@
+"""
+Torchaudio Tutorial
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+===================
+
+PyTorch is an open source deep learning platform that provides a
+seamless path from research prototyping to production deployment with
+GPU support.
+
+Significant effort in solving machine learning problems goes into data
+preparation. Torchaudio leverages PyTorch’s GPU support, and provides
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+
+
+######################################################################
+# Opening a dataset
+# -----------------
+# 
+
+
+######################################################################
+# Torchaudio supports loading sound files in the wav and mp3 format. We
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+
+plt.figure()
+plt.plot(waveform.transpose(0,1).numpy())
+
+
+######################################################################
+# Transformations
+# ---------------
+# 
+# Torchaudio supports a growing list of
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: torchaudio

File: index.rst [hunk 11]
```diff
@@ -109,11 +109,15 @@ Image
...
-.. Audio
-.. ----------------------
+Audio
+----------------------
 
-.. Uncomment below when adding content
-.. .. raw:: html
+.. customgalleryitem::
+   :figure: /_static/img/audio_preprocessing_tutorial_waveform.png
+   :tooltip: Preprocessing with Torchaudio Tutorial
```
Comment (jamarshon) [inline index.rst]: torchaudio

File: beginner_source/audio_preprocessing_tutorial.py [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+# `transformations <https://pytorch.org/audio/transforms.html>`_.
+# 
+# -  **Resample**: Resample waveform to a different sample rate.
+# -  **Spectrogram**: Create a spectrogram from a waveform.
+# -  **MelScale**: This turns a normal STFT into a Mel-frequency STFT,
+#    using a conversion matrix.
+# -  **SpectrogramToDB**: This turns a spectrogram from the
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: AmplitudeToDB

Comment (vincentqb) [reply beginner_source/audio_preprocessing_tutorial.py]: thanks

**Comment by vincentqb** [comment]
Thanks for the feedback! Updated to `torchaudio`, `AmplitudeToDB`, and `Decoding`.

File: index.rst (hunk 11 above)
Comment (jamarshon) [reply index.rst]: bump

File: beginner_source/audio_preprocessing_tutorial.py [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+# 
+
+filename = "../_static/img/steam-train-whistle-daniel_simon-converted-from-mp3.wav"
+waveform, sample_rate = torchaudio.load(filename)
+
+print("Shape of waveform: {}".format(waveform.size()))
+print("Sample rate of waveform: {}".format(sample_rate))
+
+plt.figure()
+plt.plot(waveform.transpose(0,1).numpy())
```
Comment (jamarshon) [inline beginner_source/audio_preprocessing_tutorial.py]: .t() seems cleaner than transpose(0,1)

Comment (vincentqb) [reply beginner_source/audio_preprocessing_tutorial.py]: Indeed. I updated it in other places too.

**Pull request merged**